client/jetpack-cloud/sections/agency-dashboard/sites-overview/index.tsx

```diff
@@ -46,33 +66,95 @@ export default function SitesOverview(): ReactElement {
 
 	const pageTitle = translate( 'Dashboard' );
 
+	const basePath = '/dashboard';
+
+	const navItems = [
+		{
+			key: 'all',
+			label: isMobile ? translate( 'All Sites' ) : translate( 'All' ),
+		},
+		{
+			key: 'favorites',
+			label: translate( 'Favorites' ),
+		},
+	].map( ( navItem ) => {
+		const isFavorite = navItem.key === 'favorites';
+		return {
+			...navItem,
+			count: ( isFavorite ? data?.totalFavorites : data?.total ) || 0,
+			selected: isFavorite ? filter.showOnlyFavorites : ! filter.showOnlyFavorites,
+			path: `${ basePath }${ isFavorite ? '/favorites' : '' }${ search ? '?s=' + search : '' }`,
+			onClick: () => {
+				setHightLightTab( false );
+				dispatch(
+					recordTracksEvent( 'calypso_jetpack_agency_dashboard_tab_click', {
+						nav_item: navItem.key,
+					} )
+				);
+			},
+			children: navItem.label,
+		};
+	} );
```

Comment:
This to me seems like an expensive computation and the result is a non-static value on each render which will result in further un-necessary re-renders. So, it's better to wrap it in `useMemo`.

<details>
<summary>Here is the suggestion</summary>

```suggestion
	const navItems = useMemo(
		() =>
			[
				{
					key: 'all',
					label: isMobile ? translate( 'All Sites' ) : translate( 'All' ),
				},
				{
					key: 'favorites',
					label: translate( 'Favorites' ),
				},
			].map( ( navItem ) => {
				const isFavorite = navItem.key === 'favorites';
				return {
					...navItem,
					count: ( isFavorite ? data?.totalFavorites : data?.total ) || 0,
					selected: isFavorite ? filter.showOnlyFavorites : ! filter.showOnlyFavorites,
					path: `${ basePath }${ isFavorite ? '/favorites' : '' }${ search ? '?s=' + search : '' }`,
					onClick: () => {
						setHightLightTab( false );
						dispatch(
							recordTracksEvent( 'calypso_jetpack_agency_dashboard_tab_click', {
								nav_item: navItem.key,
							} )
						);
					},
					children: navItem.label,
				};
			} ),
		[
			data?.total,
			data?.totalFavorites,
			dispatch,
			filter.showOnlyFavorites,
			isMobile,
			search,
			translate,
		]
	);
```

You of course need to import `useMemo`.
</details>

Reply:
Thanks! Fixed in 914341990cbb2cb306bbe6ba6d10040b0b954483